frontend/src/metabase/querying/components/FilterModal/FilterModal.unit.spec.tsx

```diff
@@ -0,0 +1,193 @@
+    userEvent.type(searchInput, "created");
+    act(() => jest.advanceTimersByTime(1000));
+    const sections = screen.getAllByTestId("filter-column-Created At");
+    expect(sections).toHaveLength(3);
+
+    const [ordersSection, productSection, peopleSection] = sections;
+    expect(within(ordersSection).getByText("Orders")).toBeInTheDocument();
+    expect(within(productSection).getByText("Products")).toBeInTheDocument();
+    expect(within(peopleSection).getByText("People")).toBeInTheDocument();
+
+    userEvent.click(within(ordersSection).getByText("Today"));
+    userEvent.click(within(productSection).getByText("Yesterday"));
+    userEvent.click(within(peopleSection).getByText("Last month"));
+    userEvent.click(screen.getByRole("button", { name: "Apply filters" }));
+
+    const nextQuery = getNextQuery();
+    expect(Lib.stageCount(nextQuery)).toBe(1);
+    expect(Lib.filters(nextQuery, 0)).toHaveLength(3);
+  });
+
+  it("should update existing filters", async () => {
+    const { getNextQuery } = setup({
+      query: Lib.filter(
+        query,
+        stageIndex,
+        Lib.stringFilterClause({
+          operator: "=",
+          column: findColumn("PRODUCTS", "CATEGORY"),
+          values: ["Gadget"],
+          options: {},
+        }),
+      ),
+    });
+
+    userEvent.click(screen.getByRole("tab", { name: "Product" }));
+    await waitForLoaderToBeRemoved();
+    expect(screen.getByRole("checkbox", { name: "Gadget" })).toBeChecked();
+
+    userEvent.click(screen.getByRole("checkbox", { name: "Widget" }));
+    userEvent.click(screen.getByRole("button", { name: "Apply filters" }));
+    const nextQuery = getNextQuery();
+    expect(Lib.stageCount(nextQuery)).toBe(1);
+    expect(Lib.filters(nextQuery, 0)).toHaveLength(1);
+  });
+
+  it("should remove existing filters", async () => {
+    const { getNextQuery } = setup({
+      query: Lib.filter(
+        query,
+        stageIndex,
+        Lib.stringFilterClause({
+          operator: "=",
+          column: findColumn("PRODUCTS", "CATEGORY"),
+          values: ["Gadget"],
+          options: {},
+        }),
+      ),
+    });
+
+    userEvent.click(screen.getByRole("tab", { name: "Product" }));
+    await waitForLoaderToBeRemoved();
+    expect(screen.getByRole("checkbox", { name: "Gadget" })).toBeChecked();
+
+    userEvent.click(screen.getByRole("checkbox", { name: "Gadget" }));
+    userEvent.click(screen.getByRole("button", { name: "Apply filters" }));
+    const nextQuery = getNextQuery();
+    expect(Lib.stageCount(nextQuery)).toBe(1);
+    expect(Lib.filters(nextQuery, 0)).toHaveLength(0);
+  });
+
+  it("should order columns by type", () => {
+    setup({ query });
+
+    const columns = screen.getAllByTestId(/filter-column/);
+    expect(columns).toHaveLength(9);
+    expect(within(columns[0]).getByText("Created At")).toBeInTheDocument();
+    expect(within(columns[6]).getByText("ID")).toBeInTheDocument();
```

Comment:
Can we assert the order of all 9 columns?
Asserting only 2 does not give much confidence.

Reply:
Fixed